controllers/costmanagement_controller.go

```diff
@@ -262,6 +272,83 @@ func GetAuthSecret(r *CostManagementReconciler, costInput *CostManagementInput,
 	return nil
 }
 
+func GetBodyAndHeaders(r *CostManagementReconciler, filename string) (*bytes.Buffer, *multipart.Writer) {
+	log := r.Log.WithValues("costmanagement", "GetBodyAndHeaders")
+	// set the content and content type
+	buf := new(bytes.Buffer)
+	mw := multipart.NewWriter(buf)
+	h := make(textproto.MIMEHeader)
+	h.Set("Content-Disposition", fmt.Sprintf(`form-data; name=%q; filename=%q`, "file", filename))
+	h.Set("Content-Type", "application/vnd.redhat.hccm.tar+tgz")
+	fw, err := mw.CreatePart(h)
+	f, err := os.Open(filename)
+	if err != nil {
+		log.Info("error opening file", err)
+	}
+	defer f.Close()
+	_, err = io.Copy(fw, f)
+	if err != nil {
+		log.Error(err, "Could not send request")
+	}
+	mw.Close()
+	return buf, mw
+}
+
+func Upload(r *CostManagementReconciler, costInput *CostManagementInput, method string, path string, body *bytes.Buffer, mw *multipart.Writer) (string, string, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("costmanagement", "Upload")
+	req, err := http.NewRequest(method, path, body)
+	if err != nil {
+		return "", "", err
+	}
+	// Create the header
+	if req.Header == nil {
+		req.Header = make(http.Header)
+	}
+	req = req.WithContext(ctx)
+	req.Header.Set("Content-Type", mw.FormDataContentType())
+	if costInput.Authentication == "basic" {
+		log.Info("Uploading using basic authentication!")
+		req.SetBasicAuth(costInput.BasicAuthUser, costInput.BasicAuthPassword)
+	} else {
+		log.Info("Uploading using token authentication")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", costInput.BearerTokenString))
+		req.Header.Set("User-Agent", fmt.Sprintf("cost-mgmt-operator/%s cluster/%s", costInput.OperatorCommit, costInput.ClusterID))
```

Comment:
Should we just add this no matter what the auth. Can't hurt, right?

Reply:
When I add these with basic authentication I get 401 unauthorized. When I removed them with basic auth it went back to 202s.

Reply:
ok. interesting, but I guess it is what it is.